src/core/simulation.jl

```diff
@@ -246,3 +261,12 @@ function build!(sim::Simulation; verbose::Bool = false, kwargs...)
     sim.internal.compiled_status = true
     return
 end
+
+function _check_folder(folder::String)
+    !isdir(folder) && throw(IS.ConflictingInputsError("Specified folder is not valid"))
+    testdir(folder) = try 
```

Comment:
This creates a lambda function and returns it.  Is that what you want?

Reply:
The behavior here also looks incorrect to me.